### ISSUE TITLE
Handle collision bug 

### ISSUE THREAD

**Issue by akhilkh2000** [akhilkh2000/JS-mini-projects#1]
At slower speeds of snake (300ms interval) , Faster keyboard inputs by user , results in game over.
Example - The direction of the snake is "down" and the user inputs **"left"** then **"up**" very fast ,
thus before left is updated in the coordinate of the current Head of the snake , it updates the snake to go on top and crash with itself which should not be the case

**Comment by harsha130699** [comment]
Can I contribute?

**Comment by balanaguharsha** [comment]
Isn't this issue because of the way the code is written?

**Comment by akhilkh2000** [comment]
Yes . You can make changes to it . We just need a delay so that the co-ordinates get updated before another key is registered

**Comment by thazhemadam** [comment]
Adjusting the speed at which the snake moves, and using a queue for recording keypresses before directly implementing them, might be a possible solution for this issue.
